token-client/src/main/java/com/sap/cloud/security/xsuaa/client/AbstractOAuth2TokenService.java

```diff
@@ -207,20 +217,38 @@ private OAuth2TokenResponse getOAuth2TokenResponse(@Nonnull URI tokenEndpointUri
 
 	private OAuth2TokenResponse getOrRequestAccessToken(URI tokenEndpoint, HttpHeaders headers,
 			Map<String, String> parameters) throws OAuth2ServiceException {
+		LOGGER.debug("Token was requested for endpoint uri={} with headers={} and parameters={}", tokenEndpoint,
+				headers, parameters);
 		CacheKey cacheKey = new CacheKey(tokenEndpoint, headers, parameters);
 		OAuth2TokenResponse oAuth2TokenResponse = responseCache.getIfPresent(cacheKey);
 		if (oAuth2TokenResponse == null) {
+			LOGGER.debug("Token not found in cache, requesting a new one");
 			getAndCacheToken(cacheKey);
 		} else {
+			LOGGER.debug("The token was found in cache");
 			// check if token in cache should be refreshed
 			Duration delta = getCacheConfiguration().getTokenExpirationDelta();
 			Instant expiration = oAuth2TokenResponse.getExpiredAt().minus(delta);
 			if (expiration.isBefore(Instant.now(getClock()))) {
 				// refresh (soon) expired token
+				LOGGER.debug("The cached token needs to be refreshed, requesting a new one");
 				getAndCacheToken(cacheKey);
 			}
 		}
-		return responseCache.getIfPresent(cacheKey);
+		OAuth2TokenResponse response = responseCache.getIfPresent(cacheKey);
+		logDebug(response);
+		return response;
+	}
+
+	private void logDebug(OAuth2TokenResponse response) {
+		String accessToken = response.getAccessToken();
+		String refreshToken = response.getRefreshToken();
+		if (accessToken != null) {
+			tokenLogger.logDebug(accessToken, "Access token:");
+		}
+		if (refreshToken != null) {
+			tokenLogger.logDebug(refreshToken, "Refresh token:");
```

Comment:
Yes, thanks! I just saw that the refresh_token is in another format so it can't even be logged with the `TokenLogger` Utility! It only works with JWT tokens. I will remove that part then!